# Machine Translation





## Links to Models
##### Google Translate: https://cloud.google.com/translate?hl=en & https://github.com/ahmeterenodaci/easygoogletranslate
##### Azure AI Translator: https://azure.microsoft.com/en-us/products/ai-services/ai-translator
##### MyMemory: https://mymemory.translated.net/doc/spec.php


In [1]:
import time

In [2]:
from google.colab import drive
drive.mount('/content/Drive')

Drive already mounted at /content/Drive; to attempt to forcibly remount, call drive.mount("/content/Drive", force_remount=True).


In [3]:
#DECLARING THE CONSTANTS
text = "Hello there! My name is Phil. I have been an online English tutor for almost three years now. I've been helping students from beginner to advanced level improve their English skills. I've tried teaching kids and adults. I used CPR. Look! Rewards! And many more! I would love to help you improve and make progress in learning. From phonics, grammar, pronunciation and up to conversational English. I've already earned my Tassel and Tia Felt certificate, so you're in good hands. Learn with me and let's make English a fun and easy language. See ya!"

In this code, I have translated Text from English to Spanish But it has the capability to translate in almost all languages.

##Translating Text to Spanish using Google Translate

In [4]:
# Installing the 'easygoogletranslate' library
!pip install easygoogletranslate

In [5]:
# Importing required modules
from easygoogletranslate import EasyGoogleTranslate

start_time = time.time()

translator = EasyGoogleTranslate(
    source_language='en',
    target_language='hi',
    timeout=10
)
# Enter the text to be translated
result = translator.translate(text)

print(result)
print("Google Translate : %s seconds" % (time.time() - start_time))

नमस्ते! मेरा नाम फिल है. मैं अब लगभग तीन वर्षों से ऑनलाइन अंग्रेजी ट्यूटर हूं। मैं शुरुआती से लेकर उन्नत स्तर तक के छात्रों को उनके अंग्रेजी कौशल को बेहतर बनाने में मदद कर रहा हूं। मैंने बच्चों और वयस्कों को पढ़ाने की कोशिश की है। मैंने सीपीआर का उपयोग किया। देखना! पुरस्कार! और भी कई! मुझे सीखने में सुधार करने और प्रगति करने में आपकी मदद करने में खुशी होगी। ध्वन्यात्मकता, व्याकरण, उच्चारण से लेकर बोलचाल की अंग्रेजी तक। मैंने पहले ही अपना टैसल और टिया फेल्ट प्रमाणपत्र अर्जित कर लिया है, इसलिए आप अच्छे हाथों में हैं। मेरे साथ सीखें और आइए अंग्रेजी को एक मज़ेदार और आसान भाषा बनाएं। फिर मिलेंगे!
Google Translate : 0.21838736534118652 seconds


##Translating Text to Spanish using Azure AI Translator

In [6]:
# Install the 'requests' and 'uuid' libraries
!pip install requests uuid

In [10]:
# Import required modules
import requests
import uuid

# Define your Azure Cognitive Services Translator API credentials and configuration
key = "ENTER_YOUR_API_KEY"
endpoint = "https://api.cognitive.microsofttranslator.com/"

# Define the Azure region
location = "westus2"

# Define the API path for translation
path = '/translate'
constructed_url = endpoint + path

# Define parameters
params = {
    'api-version': '3.0',
    'from': 'en',
    'to': ['hi']
}

# Define request headers
headers = {
    'Ocp-Apim-Subscription-Key': key,
    'Ocp-Apim-Subscription-Region': location,
    'Content-type': 'application/json',
    'X-ClientTraceId': str(uuid.uuid4())
}

# Define the text to be translated
body = [{
    'text': text
}]

start_time = time.time()
# Make a POST request
request = requests.post(constructed_url, params=params, headers=headers, json=body)

response = request.json()

translation = response[0]['translations'][0]['text']

print(translation)

print("Azure AI Translator: %s seconds" % (time.time() - start_time))


आपको नमस्कार! मेरा नाम फिल है। मैं लगभग तीन साल से एक ऑनलाइन अंग्रेजी ट्यूटर रहा हूं। मैं शुरुआती से उन्नत स्तर तक के छात्रों को अपने अंग्रेजी कौशल में सुधार करने में मदद कर रहा हूं। मैंने बच्चों और वयस्कों को पढ़ाने की कोशिश की है। मैंने सीपीआर का इस्तेमाल किया। देखो! पुरस्कार! और भी कई! मैं आपको सीखने में सुधार करने और प्रगति करने में मदद करना पसंद करूंगा। फोनिक्स, व्याकरण, उच्चारण और संवादात्मक अंग्रेजी तक। मैंने पहले से ही अपना टैसेल और टिया फेल्ट प्रमाण पत्र अर्जित किया है, इसलिए आप अच्छे हाथों में हैं। मेरे साथ सीखें और अंग्रेजी को एक मजेदार और आसान भाषा बनाएं। बाद में मिलेंगे!
Azure AI Translator: 0.5535500049591064 seconds


##Translating Text to Spanish using MyMemory

In [11]:
# Install the 'mtranslate' library
!pip install mtranslate

  Preparing metadata (setup.py) ... done
  Created wheel for mtranslate: filename=mtranslate-1.8-py3-none-any.whl size=3672 sha256=98d1ddb17e723ce4de909db6023e81bed833e6477ca505d92845337ce303a947
  Stored in directory: /root/.cache/pip/wheels/c2/04/15/d7654c2c4a9a52e09922967593f3278fed66059be65ca671ea
Successfully built mtranslate


In [12]:
# Import the 'mtranslate' library
from mtranslate import translate

start_time = time.time()

def translate_to_spanish(text):
    try:
        translated_text = translate(text, 'hi', 'en')
        return translated_text
    except Exception as e:
        return f"Translation error: {str(e)}"

if __name__ == "__main__":
    # Define the text to be translated
    text_to_translate = text
    translated_text = translate_to_spanish(text_to_translate)
    print(translated_text)


print("MyMemory : %s seconds" % (time.time() - start_time))

नमस्ते! मेरा नाम फिल है. मैं अब लगभग तीन वर्षों से ऑनलाइन अंग्रेजी ट्यूटर हूं। मैं शुरुआती से लेकर उन्नत स्तर तक के छात्रों को उनके अंग्रेजी कौशल को बेहतर बनाने में मदद कर रहा हूं। मैंने बच्चों और वयस्कों को पढ़ाने की कोशिश की है। मैंने सीपीआर का उपयोग किया। देखना! पुरस्कार! और भी कई! मुझे सीखने में सुधार करने और प्रगति करने में आपकी मदद करने में खुशी होगी। ध्वन्यात्मकता, व्याकरण, उच्चारण से लेकर बोलचाल की अंग्रेजी तक। मैंने पहले ही अपना टैसल और टिया फेल्ट प्रमाणपत्र अर्जित कर लिया है, इसलिए आप अच्छे हाथों में हैं। मेरे साथ सीखें और आइए अंग्रेजी को एक मज़ेदार और आसान भाषा बनाएं। फिर मिलेंगे!
MyMemory : 0.14168071746826172 seconds


### Time Taken to Translate by Different models

| Model    | Time of Execution (in sec)|
| :---        |    :----:   
| Google Translate   | 0.21    |
| Azure AI Translator | 0.55     |
| MyMemory  | 0.14        |